Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,92 +21,29 @@
 #import <vector>
 #import <CoreGraphics/CGContext.h>
 #import <CoreGraphics/CGGeometry.h>
+#import <CoreGraphics/CGDataProvider.h>
 #import <Foundation/NSData.h>
-#import <UIKit/UIImage.h>
+#import <LoggingNative.h>
+#import <CFRuntime.h>
+#import <CFBridgeUtilities.h>
+#import "D2DWrapper.h"
 #import "CGColorSpaceInternal.h"
 #import "CGImageInternal.h"
-#import "_CGLifetimeBridgingType.h"
 #import "CGSurfaceInfoInternal.h"
 
-extern "C" {
-#import <png.h>
-};
-
-// This is the format libpng expects.
-struct _RGBA_swizzle {
-    BYTE r, g, b, a;
-};
-
-// This is what comes out of pixman.
-struct _BGRA_swizzle {
-    BYTE b, g, r, a;
-};
-
-#include "LoggingNative.h"
+using namespace Microsoft::WRL;
 
 static const wchar_t* TAG = L"CGImage";
 
-@interface CGNSImage : _CGLifetimeBridgingType
-@end
-@implementation CGNSImage
-- (instancetype)copyWithZone:(NSZone*)zone {
-    return [self retain];
-}
-
-- (void)dealloc {
-#pragma diagnostic push
-#pragma clang diagnostic ignored "-Wobjc-missing-super-calls"
-    // __CGImage is a C++ class massaged into an objc object.
-    delete (__CGImage*)self;
-#pragma diagnostic pop
-}
-@end
-
-static IWLazyClassLookup _LazyUIImage("UIImage");
-
-int numCGImages = 0;
-
-__CGImage::__CGImage() {
-    numCGImages++;
-
-    _has32BitAlpha = true;
-
-#ifdef DEBUG_IMG_COUNT
-    TraceVerbose(TAG, L"Number of CGImages: %d created=%x", numCGImages, this);
-#endif
-
-    object_setClass((id)this, [CGNSImage class]);
-}
+#define RETURN_FALSE_IF_FAILED(input) RETURN_FALSE_IF(FAILED(input));
 
+// TODO #1124: remove old code
+#pragma region OLD_CODE
 static std::vector<CGImageDestructionListener> _imageDestructionListeners;
 COREGRAPHICS_EXPORT void CGImageAddDestructionListener(CGImageDestructionListener listener) {
     _imageDestructionListeners.push_back(listener);
```

Comment:
>push_back [](start = 31, length = 9)

emplace_back with an std::move is the preferred mechanism to not invoke more copies than necessary on the input objects. Not sure what a CGImageDestructionListener is yet to know how expensive a copy is but would like us to follow an emplace_back pattern where transfer of ownership is possible.